qt-models/cylindermodel.cpp

```diff
@@ -237,6 +239,28 @@ QVariant CylindersModel::data(const QModelIndex &index, int role) const
 			return static_cast<int>(cyl->type.workingpressure.mbar);
 		case SIZE_INT:
 			return static_cast<int>(cyl->type.size.mliter);
+		case SENSORS: {
+			sensorsEditable = false;
+			std::vector<int16_t> sensors(MAX_SENSORS, NO_SENSOR);
+			for (int i = 0; i < current_dc->samples; ++i) {
+				auto &sample = current_dc->sample[i];
+				for (auto s = 0; s < MAX_SENSORS; ++s) {
+					if (sample.pressure[s].mbar) {
+						if (sensors[s] == NO_SENSOR)
+							sensors[s] = sample.sensor[s];
+						else if (sensors[s] != sample.sensor[s])
+							return tr("Sensor data too complex for this UI.");
```

Comment:
If this return triggers, then `sensorEditable` is not reset to `true`. 

In #3410 I do this:
```
// Silly RAII-variable setter class: reset variable when going out of scope.
template <typename T>
struct Setter {
	T &var, old;
	Setter(T &var, T value) : var(var), old(var) {
		var = value;
	}
	~Setter() {
		var = old;
	}
};
```
and then something like
```
Setter s(sensorEditable, true);
```

Yes, I am kind of ashamed of it, but it should work.

Reply:
Actually, not setting sensorEditable to true is the intended behavior. When the data is too complex it doesn't allow editing, since it would mess things up and not be able to undo it.

I realized that I can make things work properly within the model though. So doing a rework of this that should work for all cases.

Reply:
I see - that is super-subtle, as it assumes a certain call order [data() before flags()], no?

Reply:
Ah yea, that's true. Just seemed like the natural thing and it worked when I tried it out. I was wondering if data had to be const and just made the attribute mutable while I was playing around though. I guess I either have to put similar logic in flags() or calculate and cache it in some other place.